### PR TITLE
Final clean up of code for v1.0

### PR DESCRIPTION
**Pull request by Chongjx**
Removed unused import statement.
Updated some of the prepare functions to throw proper SystemExceptions.
Changed all equals function to equalsIgnoreCase

### CODE REVIEW

**Review by R-Ramana — APPROVED** [review]
Thanks for the code clean up! LGTM

File: src/main/java/seedu/duke/data/timetable/Reminder.java [hunk 1]
```diff
@@ -51,11 +51,7 @@ public boolean equals(Object o) {
             return true;
         } else if (o instanceof Reminder) {
             Reminder r = ((Reminder) o);
-            if (r.dateToRemind.equals(dateToRemind) && r.event.getTitle().equals(event.getTitle())) {
-                return true;
-            } else {
-                return false;
-            }
+            return r.dateToRemind.equals(dateToRemind) && r.event.getTitle().equals(event.getTitle());
```
Comment (brandonywl) [inline src/main/java/seedu/duke/data/timetable/Reminder.java:54]: Will clean up naming after 1.0

**Review by brandonywl — APPROVED** [review]
LGTM!

File: src/main/java/seedu/duke/data/timetable/Reminder.java (hunk 1 above)
Comment (Chongjx) [reply src/main/java/seedu/duke/data/timetable/Reminder.java:54]: No worries

**Pull request merged**